From ff23aa8b498da625b5c9bffe18e38b4bfcd3e873 Mon Sep 17 00:00:00 2001
From: Maher Khenissi <maherx.khenissi@intel.com>
Date: Wed, 21 Nov 2012 10:23:19 +0100
Subject: [PORT FROM MAIN] calibrator: fix klocwork issues.

BZ: 67389

function strncpy may incorrectly check buffer boundaries.

Change-Id: Iac7c9dd8765fdefbbb327c65b0f704ab6ec51677
Orig-Change-Id: Ib903b1618538f0fd5316a2a02e9daf3aecd7c953
Signed-off-by: Maher Khenissi <maherx.khenissi@intel.com>
Reviewed-on: http://android.intel.com:8080/75162
Reviewed-by: Casasnovas, QuentinX <quentinx.casasnovas@intel.com>
Reviewed-by: Poulain, LoicX <loicx.poulain@intel.com>
Reviewed-by: De Cesco, Jonathan <jonathan.de.cesco@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Reviewed-by: Garcia, JeremieX <jeremiex.garcia@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Reviewed-on: http://android.intel.com:8080/78012
Reviewed-by: Li, XiaojingX <xiaojingx.li@intel.com>
Tested-by: Li, XiaojingX <xiaojingx.li@intel.com>
---
 calibrator/plt.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/calibrator/plt.c b/calibrator/plt.c
index b430abe..5ee03a3 100644
--- a/calibrator/plt.c
+++ b/calibrator/plt.c
@@ -112,7 +112,7 @@ static int isiffup(const char *vname)
 
 	if (skfd) {
 		struct ifreq ifr;
-		strncpy (ifr.ifr_name, vname, sizeof ifr.ifr_name);
+		strncpy (ifr.ifr_name, vname, sizeof(ifr.ifr_name)-1);
 		ifr.ifr_name[sizeof(ifr.ifr_name)-1]='\0';
 		if (!ioctl (skfd, SIOCGIFFLAGS, &ifr)) {
 		   const short int flags = ifr.ifr_flags;
@@ -130,7 +130,7 @@ static int setiffdown(const char *vname)
 
 	if (skfd) {
 		struct ifreq ifreq;
-		strncpy(ifreq.ifr_name, vname, sizeof(ifreq.ifr_name));
+		strncpy(ifreq.ifr_name, vname, sizeof(ifreq.ifr_name)-1);
 		ifreq.ifr_name[sizeof(ifreq.ifr_name)-1]='\0';
 		if (!ioctl (skfd, SIOCGIFFLAGS, &ifreq)) {
 			short int flags = ifreq.ifr_flags;
-- 
1.7.4.1

